src/main/java/org/broadinstitute/hellbender/tools/exome/NormalizeBySampleDepth.java

```diff
@@ -0,0 +1,124 @@
+package org.broadinstitute.hellbender.tools.exome;
+
+import org.apache.commons.collections4.list.SetUniqueList;
+import org.apache.commons.math3.linear.DefaultRealMatrixChangingVisitor;
+import org.apache.commons.math3.linear.DefaultRealMatrixPreservingVisitor;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.CopyNumberProgramGroup;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Calculates basic statistics for coverage per targets and per sample.
+ * <p>
+ *   This tool takes a read counts file and outputs a read counts file in which coverage at each target
+ *   and sample is divided by the average coverage of that sample.
+ * </p>
+ * <p>
+ *   The input and output format for the coverage file is described in {@link ReadCountCollectionUtils}.
+ * </p>
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        summary = "Normalizes read counts by samples' average coverage.",
+        oneLineSummary = "Normalizes read counts by samples' average coverage.",
+        programGroup = CopyNumberProgramGroup.class
+)
+public class NormalizeBySampleDepth extends CommandLineProgram {
+    public static final String WEIGHTED_AVERAGE_LONG_NAME = "weightedAverage";
+    public static final String WEIGHTED_AVERAGE_SHORT_NAME = "w";
+
+    @Argument(
+            doc = "Input coverage file",
+            fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File inputFile;
+
+    @Argument(
+            doc = "Output coverage file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File outputFile;
+
+    @Argument(
+            doc = "Weighted average with weights proportional to target sizes",
+            fullName = WEIGHTED_AVERAGE_LONG_NAME,
+            shortName = WEIGHTED_AVERAGE_SHORT_NAME,
+            optional = true
+    )
+    protected boolean weightByTargetSize = false;
+
+    @Override
+    public Object doWork() {
+        final ReadCountCollection inputCounts;
+        try {
```

Comment:
The bulk of doWork belongs in another class.  We should get in the habit of separating the CLI (presentation) layer of code from backend classes.  I can explain why this is important in person.  

The general structure should be:
```
@Override
public Object doWork() {
  // read file
  // call backend object to the processing on a ReadCountCollection, which returns the counts or some class of results.
  // write to file
}
```